Add E2E tests for st.code copy-to-clipboard functionality

sfc-gh-lwilby opened the pull request:
## Describe your changes

Adds E2E tests for the `st.code` copy-to-clipboard functionality to verify expected behavior in response to GitHub Issue #12792, where users reported the copy button was not working properly.

**Changes Made:**

- Added `test_copy_to_clipboard_functionality`: Verifies single-line code copying works correctly
- Added `test_copy_to_clipboard_multiline_code`: Verifies multiline code blocks (with newlines and indentation) copy correctly
- Tests use Playwright's clipboard API to verify actual clipboard contents
- Tests verify copy button visibility on hover and state change after successful copy

## GitHub Issue Link (if applicable)

Related to GitHub Issue #12792

## Testing Plan

- [x] E2E Tests - Two new tests added to verify copy-to-clipboard behavior
- [x] Tests use clipboard permissions and navigator.clipboard.readText() to verify actual clipboard contents
- [x] Tests verify both single-line and multiline code preservation

**Additional Notes:**

These tests document the expected behavior of the copy-to-clipboard feature and can serve as regression tests to catch any future issues with this functionality.

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

sfc-gh-lwilby committed:
Add E2E tests for st.code copy-to-clipboard functionality

Tests verify clipboard behavior for single-line and multiline code blocks.
Related to GitHub Issue #12792.

Co-authored-by: lawilby <laura.wilby+oss@snowflake.com>

sfc-gh-lwilby committed:
Use @pytest.mark.only_browser decorator for clipboard tests

Replace inline browser skip logic with conventional @pytest.mark.only_browser
decorator for st.code clipboard tests. Tests now follow project conventions
and skip gracefully on Firefox/WebKit where Playwright doesn't support
clipboard permissions.

- test_copy_to_clipboard_functionality: Chromium only
- test_copy_to_clipboard_multiline_code: Chromium only

Fixes CI failures on Firefox (clipboard-read) and WebKit (clipboard-write).

Related: gh-12792

Co-authored-by: lawilby <laura.wilby+oss@snowflake.com>

mayagbarnes reviewed (approved):
LGTM! 
Think there are just some nits in graphite comments.
Re: Copilot comments to add helper - seems a bit unnecessary to me but up to you!

sfc-gh-nbellante commented on e2e_playwright/st_code_test.py:
```diff
@@ -211,3 +212,83 @@ def test_container_with_code_blocks(app: Page, assert_snapshot: ImageCompareFunc
+    # Wait for button to be visible
+    expect(copy_button).to_be_visible()
+
+    # Click the copy button
```
suggestion (nit): some of these llm comments are a bit unnecessary

sfc-gh-nbellante commented on e2e_playwright/st_code_test.py:
```diff
@@ -211,3 +212,83 @@ def test_container_with_code_blocks(app: Page, assert_snapshot: ImageCompareFunc
+    clipboard_text = app.evaluate("async () => await navigator.clipboard.readText()")
+
+    # Expected multiline code
+    expected_code = textwrap.dedent(
```
observation: TIL! That's pretty cool and would've been helpful in the public notebooks project 😅

sfc-gh-lwilby committed:
Address PR review comments

Replace assert with wait_until for clipboard verification to reduce
flakiness, and remove unnecessary LLM-style comments.

Made-with: Cursor

Co-authored-by: lawilby <laura.wilby+oss@snowflake.com>
Made-with: Cursor
